adoptingerlang.org

```diff
@@ -3405,27 +3542,269 @@ $ kubectl apply -k deployment/overlays/dev
+
+Since the =ttlSecondsAfterFinished= options is still an alpha feature and requires manually enabling. You can try the feature on Google Cloud by creating an [[https://cloud.google.com/kubernetes-engine/docs/concepts/alpha-clusters][alpha cluster]], but these clusters can only live for 30 days. An alternative solution is to manually (or with a script that can run in a CI pipeline) run migrations and delete the =Job= after completing a deployment.
+
+An advantage of running the =Job= separately from the main =kubectl apply= is the deployment can be stopped if the =Job= failed.
+
+**** Init Containers
+
+#+BEGIN_SRC yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: service-discovery
+spec:
+  replicas: 1
+  template:
+    spec:
+      volumes:
+      - name: migrations
+        emptyDir:
+          medium: Memory
+      initContainers:
+      - args:
+        - "-url=jdbc:postgresql://$(POSTGRES_SERVICE):5432/discovery"
+        - -user=discovery
+        - -password=password
+        - -connectRetries=60
+        - validate
+        image: boxfuse/flyway:5.2.4
+        name: flyway-validate
+        volumeMounts:
+        - name: migrations
+          mountPath: /flyway/sql
+
+      - name: service-discovery-sql
+        image: service_discovery-sql
+        volumeMounts:
+          - name: migrations
+            mountPath: /flyway/sql
+        command: ["/bin/sh"]
+        args: ["-c", "cp /opt/service_discovery/sql/* /flyway/sql"]
+#+END_SRC
+
 *** Tilt for Local Development
 
+[[https://tilt.dev/][Tilt]] is a tool for deploying and updating Docker images and Kubernetes deploys locally for development purposes. The easiest way to get started with Tilt is using a registry within the local Kubernetes cluster. For =microk8s= a registry can be enabled easily, we will need DNS as well so enable that at this time as well:
+
 #+BEGIN_SRC shell
 $ microk8s.enable registry
 $ microk8s.enable dns
 #+END_SRC
 
-=/etc/docker/daemon.json=
+To allow the host Docker to publish to this registry it must be added to =/etc/docker/daemon.json=:
 
 #+BEGIN_SRC json
 {
-  ...
+  [...]
   "insecure-registries" : ["localhost:32000"]
-  ...
+  [...]
 }
 #+END_SRC
 
-*** Database Migrations
-**** Jobs
-**** Init Containers
-*** StatefulSet
+Tilt works off of a file named =Tiltfile= at the root of the project. Stepping through the =Tiltfile= at the root of [[https://github.com/adoptingerlang/service_discovery][service_discovery]] it begins with:
+
+#+BEGIN_SRC python
+allow_k8s_contexts("microk8s")
+default_registry('127.0.0.1:32000')
+#+END_SRC
+
+=allow_k8s_contexts= restricts the Kubernetes context that Tilt will run against to =microk8s=, this is to protect the user from accidentally deploying their local development resources and images to an external Kubernetes cluster, like production! =default_registry= sets where to push Docker images built by Tilt, in this case the registry enabled in =microk8s= is used.
```

Comment:
```suggestion
=allow_k8s_contexts= restricts the Kubernetes context that Tilt will run against to =microk8s=. This protects the user from accidentally deploying their local development resources and images to an external Kubernetes cluster, such as production. =default_registry()= sets where to push Docker images built by Tilt, in this case the registry enabled in =microk8s= is used.
```

Reply:
Outdated, I changed the wording here because it is now the default way Tilt works.